pkg/certificates/reconciler/certificates.go

```diff
@@ -95,16 +105,29 @@ func (r *reconciler) ReconcileKind(ctx context.Context, secret *corev1.Secret) p
 	cert, _, err := parseAndValidateSecret(secret, true)
 	if err != nil {
```

Comment:
As updating the certificate only when `parseAndValidateSecret()` failed, this library does not assume certificate will be reconciled when secret's `RoutingId` is changed. It is alright, isn't it?

Reply:
well... someone may change the `RoutingId` in the Secret and we should reconcile - so I think this requires fixing...

Reply:
Added validation of sans in parseAndValidateSecret